File: week_09/yumyum/container.js
var maxArea = function (height) {
  let answer = 0;

  let left = 0;
  let right = height.length - 1;

  while (left < right) {
    let h = Math.min(height[left], height[right]);
    answer = Math.max(answer, h * (right - left));
    if (height[left] < height[right]) {
      left++;
    } else {
      right--;
    }
  }

  return answer;
};

File: week_09/comet/container.kt
import kotlin.math.max
import kotlin.math.min

class Solution {
    fun maxArea(height: IntArray): Int {
        var answer = 0
        
        var start = 0
        var end = height.size - 1
        
        while (start < end) {
            val sHeight = height[start]
            val eHeight = height[end]
            val area = (end - start) * min(sHeight, eHeight)
            answer = max(answer, area)
        
            if (sHeight > eHeight) end--
            else start++
        }
        
        return answer
    }
}